libs/blocks/caas/utils.js

```diff
@@ -16,17 +16,57 @@ const fetchWithTimeout = async (resource, options = {}) => {
   return response;
 };
 
-export const loadStrings = async (url) => {
-  // TODO: Loc based loading
+const pageConfig = pageConfigHelper();
+const pageLocales = Object.keys(pageConfig.locales || {});
+
+export function getPageLocale(currentPath, locales = pageLocales) {
+  const possibleLocale = currentPath.split('/')[1];
+  for (let i = 0; i < locales.length; i += 1) {
+    const locale = locales[i];
+    if (locale === possibleLocale) {
+      return locale;
+    }
+  }
+  // defaults to en_US
+  return '';
+}
+
+export const loadStrings = async (
+  url,
+  pathname = window.location.pathname,
+  locales = pageLocales,
+) => {
   if (!url) return {};
-  const resp = await fetch(url);
-  if (!resp.ok) return {};
-  const json = await resp.json();
-  const convertToObj = (data) => data.reduce((obj, { key, val }) => {
-    obj[key] = val;
-    return obj;
-  }, {});
-  return convertToObj(json.data);
+  try {
+    const locale = getPageLocale(pathname, locales);
+    let localizedURL = new URL(url);
+    if (locale) {
+      localizedURL.pathname = `${locale}${localizedURL.pathname}`;
+      localizedURL = localizedURL.toString();
+    }
+    let resp = await fetch(`${localizedURL}.plain.html`);
+    if (!resp.ok) {
+      resp = await fetch(`${url}.plain.html`);
+    }
+    if (!resp.ok) {
+      return {};
+    }
+    const ans = {};
+    const html = await resp.text();
+    const parser = new DOMParser();
+    const document = parser.parseFromString(html, 'text/html');
+    const nodes = document.querySelectorAll('.string-mappings > div');
+    for (let i = 0; i < nodes.length; i += 1) {
+      const parent = nodes[i];
+      const children = parent.querySelectorAll('div');
+      const key = children[0].innerText;
+      const val = children[1]?.innerHTML;
+      ans[key] = val || '';
+    }
```

Comment:
If possible, try to avoid for loops and use higher-order methods
```suggestion
    return [...nodes].reduce((ans, parent) => {
      const children = parent.querySelectorAll('div');
      const key = children[0]?.innerText;
      const val = children[1]?.innerHTML;
      if (key) {
        ans[key] = val || '';
      }
      return ans;
    }, {});
```